addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1778,6 +1778,16 @@ const ColorpickerUserValueWidget = SelectUserValueWidget.extend({
             options.getTemplate = wysiwyg.getColorpickerTemplate.bind(wysiwyg);
         }
         this.colorPaletteWrapper?.destroy();
+        const sidebarDocument = this.colorPaletteEl.ownerDocument; 
+        if (!(this.colorPaletteEl instanceof sidebarDocument.defaultView.HTMLElement)) {
+            // When inside an iframe, the element for mounting a component must
+            // be an instance of the iframe's HTMLElement, or else target
+            // validation for attachComponent fails.
+            const newEl = sidebarDocument.importNode(this.colorPaletteEl, true);
```

Comment:
I looked up `importNode` because I'd never seen it before. I [learned](https://stackoverflow.com/a/41322735) that `cloneNode` should work too because inserting a node in a document is supposed to make the document adopt the node, so that the owner document of the node before it's in the DOM may be the wrong one, but once it's inserted it should be correct. That makes me wonder how come the document was wrong here. Do you have any idea?
Not taking away from the fix, I do think this is the right way to go about it. But I'm curious.

Reply:
Clone does not work, because it clones (!) the original node, with the same prototype of HTMLElement.
The problem here is not the `ownerDocument` property. Indeed the browser takes care of that when inserting nodes into a DOM (an implicit `adopt` is done).
![image](https://github.com/odoo/odoo/assets/65516503/c19f82d9-0f09-4123-a0c0-85cee0139f99)

`el instanceof el.ownerDocument.defaultView.Node` evaluated to `false` basically means that this element was not created by the document it is currently a child of.